.gitignore

```diff
@@ -23,6 +23,22 @@
 hs_err_pid*
 /bin/
 
-# Build-system generated directories
-/build/
-/dist/
+<<<<<<< HEAD
+# Build Folders
+build/
+dist/
+
+# code populated files
+database/**
+benchmarks/**
+suitablePrgms/**
+!./suitablePrgms/moloneycharles/testclasses/*
+# Use '!' to designate and list Files to be included as needed in these dirs
```

Comment:
This PR seems reasonable overall.

Since most of the deleted files which are generated/downloaded, I am *mostly* okay with this.  However, because projects are deleted from GitHub often enough, I'm concerned that these tests may disappear.  Is there a plan to ensure that we have an archive copy of the benchmarks used?

Additionally, I think following the second option mentioned below should be done as part of this commit.  The README in root should describe the directories, and hidden files used instead.

Reply:
One of the goals, to my understanding, is to begin compiling another repository of Benchmarks that are created and verified. That repository will be separate from this one to avoid bloat in the source code and simplify PRs in the future. Some of the benchmarks should be added there soon, such as the ones that @rubalgoyal did.

As for existing copies at this time. We have the version control system so if we do have need to go back in time and retrieve the necessary benchmarks or repositories the ones that have already been saved/added to github should be available still.

Reply:
I think you are right in that the information in the read me files should be in the main one and not here. As for the directories themselves, after looking more closely at the code the directories: benchmarks, database and suitablePrgms seem to also created by the code so there should be no need to track these directories at all. This makes the README.md's redundant and hidden files unnecessary. Can you confirm that these directories are created and not simply hard coded into the paths? If so I will remove the readme and hidden files and allow the code to do its job.

Reply:
I can confirm.